tests/test_bug_features.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import os
+
+from bugbug.bug_features import component
+from bugbug.bug_features import has_crash_signature
+from bugbug.bug_features import has_github_url
+from bugbug.bug_features import has_regression_range
+from bugbug.bug_features import has_str
+from bugbug.bug_features import has_url
+from bugbug.bug_features import has_w3c_url
+from bugbug.bug_features import is_coverity_issue
+from bugbug.bug_features import keywords
+from bugbug.bug_features import landings
+from bugbug.bug_features import patches
+from bugbug.bug_features import product
+from bugbug.bug_features import severity
+from bugbug.bug_features import title
+from bugbug.bug_features import whiteboard
+
+
+def read(path, function):
+    check = function()
+    if not os.path.exists(path):
+        return ()
+
+    data = []
+    with open(path, 'r') as f:
+        for line in f:
+            data.append(check(json.loads(line)))
+
+    return data
+
+
+def test_has_str():
+
+    bugs = read('tests/features/has_str.json', has_str)
+    results = ['yes', None]
+    assert results == bugs
+
+
+def test_has_regression_range():
+
+    bugs = read('tests/features/has_regression_range.json', has_regression_range)
+    results = ['yes', None]
```

Comment:
Nit: here and in the other test functions, you can avoid having the `results` variable and just do `assert [...] == bugs`.

Reply:
Or even better, make read do the assertions, by passing the expected values to it.